mitosheet/mitosheet/step_performers/ai_transformation.py

```diff
@@ -0,0 +1,81 @@
+
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+
+import pandas as pd
+from time import perf_counter
+from typing import Any, Dict, List, Optional, Set, Tuple
+from mitosheet.code_chunks.code_chunk import CodeChunk
+from mitosheet.code_chunks.ai_transformation_code_chunk import AITransformationCodeChunk
+
+from mitosheet.state import State
+from mitosheet.step_performers.step_performer import StepPerformer
+from mitosheet.step_performers.utils import get_param
+from mitosheet.ai.recon import exec_and_get_new_state_and_last_line_expression_value
+
+class AITransformationStepPerformer(StepPerformer):
+    """
+    Allows you to execute an arbitrary chunk of code, generated by an AI.
+    """
+
+    @classmethod
+    def step_version(cls) -> int:
+        return 1
+
+    @classmethod
+    def step_type(cls) -> str:
+        return 'ai_transformation'
+
+    @classmethod
+    def execute(cls, prev_state: State, params: Dict[str, Any]) -> Tuple[State, Optional[Dict[str, Any]]]:
+        user_input: str = get_param(params, 'user_input')
+        prompt_version: str = get_param(params, 'prompt_version')
+        prompt: str = get_param(params, 'prompt')
+        completion: str = get_param(params, 'completion')
+        edited_completion: str = get_param(params, 'edited_completion')
+
+        pandas_start_time = perf_counter()
+        post_state, last_line_value = exec_and_get_new_state_and_last_line_expression_value(prev_state, edited_completion)
```

Comment:
`exec_and_get_new_state_and_last_line_expression_value` should probably return the post_state and the entire recon object because we want the step performer to return the recon data to the frontend as the result when its called. 

If we do this, then we can also fill in `get_modified_dataframe_indexes` and `get_created_dataframe_indexes` which the results of our recon

Reply:
I change this in the next PR -- good callout here. Sadly, we can't actually use this in the get_modified_dataframe_indexes and get_created_dataframe_indexes unless we refactor them to take state or execution data....